sdk/src/main/java/me/uport/sdk/Uport.kt

```diff
@@ -133,24 +109,26 @@ object Uport {
      *
      * To really create a new account, call [deleteAccount] first.
      */
-    suspend fun createAccount(networkId: String, seedPhrase: String? = null): Account {
+    suspend fun createAccount(networkId: String, seedPhrase: String? = null): HDAccount {
         if (!initialized) {
             throw UportNotInitializedException()
         }
 
         val newAccount = if (seedPhrase.isNullOrBlank()) {
             accountCreator.createAccount(networkId)
-        } else {
+        }
+        else {
             accountCreator.importAccount(networkId, seedPhrase)
         }
         accountStorage?.upsert(newAccount)
-        defaultAccount = defaultAccount ?: newAccount
-        val result = if (newAccount.handle == defaultAccount?.handle) {
-            defaultAccount ?: newAccount
-        } else {
-            newAccount
+
+        if (accountStorage?.getDefaultAccount() == null) {
```

Comment:
This block is redundant because the accountStorage already gets updated when the default account is assigned.

The next line of code already takes care of storage in the `defaultAccount` setter method.
```kotlin
defaultAccount = defaultAccount ?: newAccount
```